ext/socket/raddrinfo.c

```diff
@@ -10,6 +10,22 @@
 
 #include "rubysocket.h"
 
+// GETADDRINFO_IMPL == 0 : call getaddrinfo/getnameinfo directly
+// GETADDRINFO_IMPL == 1 : call getaddrinfo/getnameinfo without gvl (but uncancellable)
+// GETADDRINFO_IMPL == 2 : call getaddrinfo/getnameinfo in a dedicated pthread
+//                         (and if the call is interrupted, the pthread is detached)
+
+#ifndef GETADDRINFO_IMPL
+#  ifdef GETADDRINFO_EMU
+#    define GETADDRINFO_IMPL 0
+#  elif !defined(HAVE_PTHREAD_CREATE) || !defined(HAVE_PTHREAD_DETACH) || defined(__MINGW32__) || defined(__MINGW64__)
```

Comment:
if `!defined(HAVE_PTHREAD_CREATE) || !defined(HAVE_PTHREAD_DETACH)` is given, both functions are available.
You can also use `#define HAVE_PTHREAD_H 1` in `.ext/include/x86_64-linux/ruby/config.h` (arch).

Reply:
I was worried about environments that only provide one of the two, or environments where pthread.h are available but not actually usable. So I thought this would be a safe bet. I may try it after Ruby CI passes.